core/src/test/java/io/undertow/server/handlers/ChunkedResponseTransferCodingTestCase.java

```diff
@@ -75,7 +80,36 @@ public void handleRequest(final HttpServerExchange exchange) {
                     throw new RuntimeException(e);
                 }
             }
-        });
+        })
+                .addExactPath("/buffers", new HttpHandler() {
+                    @Override
+                    public void handleRequest(HttpServerExchange exchange) throws Exception {
+                        ByteBuffer[] buffers = new ByteBuffer[] {
+                                ByteBuffer.wrap("prefix".getBytes(StandardCharsets.UTF_8)),
+                                ByteBuffer.wrap("hello, ".getBytes(StandardCharsets.UTF_8)),
+                                ByteBuffer.wrap("world".getBytes(StandardCharsets.UTF_8)),
+                                ByteBuffer.wrap("suffix".getBytes(StandardCharsets.UTF_8))
+                        };
+                        int offset = 1;
+                        int length = 2;
+                        StreamSinkChannel channel = exchange.getResponseChannel();
+                        Channels.writeBlocking(channel, buffers, offset, length);
+                        exchange.endExchange();
+                    }
+                }));
+    }
+
+    @Test
+    public void testMultiBufferWrite() throws IOException {
+        HttpGet get = new HttpGet(DefaultServer.getDefaultServerURL() + "/buffers");
+        TestHttpClient client = new TestHttpClient();
+        try {
+            HttpResponse result = client.execute(get);
```

Comment:
Without the fix, this test times out as the handler continues attempting to write buffers that ChunkedStreamSinkConduit doesn't drain, as it only looks at the buffer at index 1, where buffers at index 1 and 2 should be written.